### DISCUSSION TITLE
Will Metrics ever become production ready?

### DISCUSSION THREAD

**Discussion by acromc** [open-telemetry/opentelemetry-go#3119] [Q&A]
Hi. I find it weird that metrics are still labeled alpha after 3 years of development for a language that is probably the de facto standard of modern microservices. Is there any hope for open telemetry, especially metrics, to ever become production ready for golang? Thanks.

**Comment by dmathieu** [comment]
Hi,

Implementation for the metrics API is done.
The SDK is in progress in the `new_sdk/main` branch, which you're welcome to track and contribute to.

**Reply by acromc** [reply]
Thanks for the quick response. Does that mean that using the current API in the master branch safe to use for now or wait until `new_sdk/main` gets merged if it is too different?

**Reply by dmathieu** [reply]
That branch is a WIP. It's not meant for production use, and can break without notice.

**Reply by acromc** [reply]
Thank you @dmathieu . I was actually talking about the `master` branch itself. Is it safe to use in production for metrics today? I don't mind about slight changes in the API here and there in the future. I am just asking if it's robust enough to be used in production.

**Reply by dmathieu** [reply]
The main branch can be used in production today. But you should expect major breaking changes once the new sdk branch is merged.

**Reply by acromc** [reply]
I don't mind breaking changes as long as the current main branch is robust enough for production use. Thank you @dmathieu !

**Reply by Aneurysm9** [reply]
The metrics API is still pre-1.0, so we do not guarantee that there will not be breaking changes in the API, but we believe that it is unlikely to change significantly as it implements the stable API specification.
